### PR TITLE
Fix nightly build workflow

### PR DESCRIPTION
**Pull request by MetRonnie**
Nightly builds were supposed to be dropped from commit history, but weren't.

Problem was `git rebase -i --keep-base` wasn't getting any commits. Also need to force push.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -64,11 +64,12 @@ jobs:
...
           pip install './cylc-flow[all]'
 
-      - name: checkout gh-pages
-        uses: actions/checkout@v2
+      - name: checkout gh-pages branch
+        uses: actions/checkout@v2.3.1
         with:
           ref: gh-pages
           path: gh-pages
+          fetch-depth: 0 # all history (unfortunately all branches too)
```
Comment (oliver-sanders) [inline .github/workflows/nightly.yml:72]: Derp, of course it was only ever working with a single commit before :/

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -81,17 +82,20 @@ jobs:
...
 
       - name: remove previous nightly build
         run: |
           cd gh-pages
+          echo "INFO: gh-pages branch at $(git rev-parse HEAD) pre-rebase"
+          echo "Use git fetch <upstream> <commit> if you need to recover the branch"
+          echo ""
           EDITOR='sed -i "s/pick \([0-9a-z]* -nightly build-\)/drop \1/"' \
-            git rebase -i --keep-base
+            git rebase -i gh-pages.root
```
Comment (oliver-sanders) [inline .github/workflows/nightly.yml:95]: This doesn't work for me, how about `git rebase -i --root`?

**Comment by oliver-sanders** [comment]
Removed the surplus nightly builds to give this action a chance of passing when it's merged.

**Comment by MetRonnie** [comment]
> Removed the surplus nightly builds to give this action a chance of passing when it's merged.

Actually the rebase in the workflow would have taken care of all of them anyway

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (MetRonnie) [reply .github/workflows/nightly.yml:95]: Ah, should be `git rebase -i origin/gh-pages.root` (I created that branch)

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -81,17 +82,20 @@ jobs:
...
 
       - name: remove previous nightly build
         run: |
           cd gh-pages
+          echo "INFO: gh-pages branch at $(git rev-parse HEAD) pre-rebase"
+          echo "Use git fetch <upstream> <commit> if you need to recover the branch"
+          echo ""
           EDITOR='sed -i "s/pick \([0-9a-z]* -nightly build-\)/drop \1/"' \
-            git rebase -i --keep-base
+            git rebase -i origin/gh-pages.root
```
Comment (oliver-sanders) [inline .github/workflows/nightly.yml:95]: I'm not convinced by this `gh-pages.root` approach, it's a potential breakage.

Comment (MetRonnie) [reply .github/workflows/nightly.yml:95]: Ok, then just use `--root`?

Comment (oliver-sanders) [reply .github/workflows/nightly.yml:95]: Fine by me, probs not worth trying to work out the previous commit hash for rebase.

**Comment by oliver-sanders** [comment]
Test failure is due to the (ever flaky) tutorial suite so can be ignored.

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -109,6 +113,8 @@ jobs:
           git -C gh-pages add "$build_name" 'versions.json'
 
       - name: push changes
+        working-directory: gh-pages
         run: |
-          git -C gh-pages commit -m '-nightly build-'
-          git -C gh-pages push origin HEAD
+          git status
+          git commit -m '-nightly build-'
+          git push --force-with-lease origin gh-pages
```
Comment (oliver-sanders) [inline .github/workflows/nightly.yml:120]: `--force-with-lease`, had to google that one.

**Comment by oliver-sanders** [comment]
Happy to slide this one in with one approval as it is a nightly build of an undeployed system, let's see it work.

**Pull request merged**